x-pack/plugins/streams_app/public/components/stream_detail_schema_editor/index.tsx

```diff
@@ -4,15 +4,131 @@
-import React from 'react';
-
-export function StreamDetailSchemaEditor({
-  definition: _definition,
-  refreshDefinition: _refreshDefinition,
-}: {
-  definition?: StreamDefinition;
+import React, { useEffect, useState } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiProgress,
+  EuiSearchBar,
+  EuiSpacer,
+  EuiPortal,
+  Query,
+} from '@elastic/eui';
+import { ReadStreamDefinition } from '@kbn/streams-plugin/common';
+import { useEditingState } from './hooks/use_editing_state';
+import { SchemaEditorFlyout } from './flyout';
+import { useKibana } from '../../hooks/use_kibana';
+import { useUnpromotingState } from './hooks/use_unpromoting_state';
+import { SimpleSearchBar } from './simple_search_bar';
+import { UnpromoteFieldModal } from './unpromote_field_modal';
+import { useStreamsAppFetch } from '../../hooks/use_streams_app_fetch';
+import { FieldsTableContainer } from './fields_table';
+
+interface SchemaEditorProps {
+  definition?: ReadStreamDefinition;
   refreshDefinition: () => void;
-}) {
-  return <>{'TODO'}</>;
+  isLoadingDefinition: boolean;
+}
+
+export function StreamDetailSchemaEditor(props: SchemaEditorProps) {
+  if (!props.definition) return null;
+  return <Content definition={props.definition} {...props} />;
 }
+
+const Content = ({
+  definition,
+  refreshDefinition,
+  isLoadingDefinition,
+}: Required<SchemaEditorProps>) => {
+  const {
+    dependencies: {
+      start: {
+        streams: { streamsRepositoryClient },
+      },
+    },
+    core: {
+      notifications: { toasts },
+    },
+  } = useKibana();
+
+  const [query, setQuery] = useState<Query | undefined>(EuiSearchBar.Query.MATCH_ALL);
+
+  const {
+    value: unmappedFieldsValue,
+    loading: isLoadingUnmappedFields,
+    refresh: refreshUnmappedFields,
+  } = useStreamsAppFetch(
+    ({ signal }) => {
+      return streamsRepositoryClient.fetch('GET /api/streams/{id}/schema/unmapped_fields', {
+        signal,
+        params: {
+          path: {
+            id: definition.id,
+          },
+        },
+      });
+    },
+    [definition.id, streamsRepositoryClient]
+  );
+
+  const editingState = useEditingState({
+    definition,
+    streamsRepositoryClient,
+    refreshDefinition,
+    refreshUnmappedFields,
+    toastsService: toasts,
+  });
+
+  const unpromotingState = useUnpromotingState({
+    definition,
+    streamsRepositoryClient,
+    refreshDefinition,
+    refreshUnmappedFields,
+    toastsService: toasts,
+  });
+
+  const { reset } = editingState;
+
+  // If the definition changes (e.g. navigating to parent stream), reset the entire editing state.
+  useEffect(() => {
+    reset();
+  }, [definition.id, reset]);
+
+  return (
+    <EuiFlexGroup direction="column">
+      {isLoadingDefinition || isLoadingUnmappedFields ? (
+        <EuiPortal>
+          <EuiProgress size="xs" color="accent" position="fixed" />
+        </EuiPortal>
+      ) : null}
+      <EuiSpacer size="l" />
```

Comment:
Why the spacer here? - seems like we can take that screen space back for the table.